pkg/serviceaccount/openidmetadata.go

```diff
@@ -44,26 +47,68 @@ const (
 	JWKSPath = "/openid/v1/jwks"
 )
 
-// OpenIDMetadata contains the pre-rendered responses for OIDC discovery endpoints.
-type OpenIDMetadata struct {
-	ConfigJSON       []byte
-	PublicKeysetJSON []byte
+// OpenIDMetadataProvider returns pre-rendered responses for OIDC discovery endpoints.
+type OpenIDMetadataProvider interface {
+	GetConfigJSON() (json []byte, maxAge int, err error)
+	GetKeysetJSON() (json []byte, maxAge int, err error)
+}
+
+type openidConfigProvider struct {
+	issuerURL, jwksURI string
+	pubKeyGetter       PublicKeyGetter
+	config             atomic.Pointer[openidConfig]
+}
+type openidConfig struct {
+	configJSON []byte
+	keysetJSON []byte
 }
 
-// NewOpenIDMetadata returns the pre-rendered JSON responses for the OIDC discovery
+func (p *openidConfigProvider) GetConfigJSON() ([]byte, int, error) {
+	return p.config.Load().configJSON, p.pubKeyGetter.GetCacheAgeMaxSeconds(), nil
```

Comment:
Mostly just thinking out loud: are we expecting `GetCacheAgeMaxSeconds` to change over time?

Reply:
I don't think so